src/bms/lex.rs

```diff
@@ -4,32 +4,70 @@ pub mod command;
 mod cursor;
 pub mod token;
 
+use std::borrow::Cow;
+
 use thiserror::Error;
 
+use crate::lex::command::channel::{Channel, read_channel_beat};
+
 use self::{cursor::Cursor, token::Token};
 
+/// A position in the text.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct TextPosition {
+    /// The line number of the position.
+    pub line: usize,
+    /// The column number of the position.
+    pub col: usize,
+}
```

Comment:
This struct name may be confused because it's hard to understand what `Text` means. How about `SourcePosition`?